uast/utils.go

```diff
@@ -17,3 +17,57 @@ func Tokens(n *Node) []string {
 	}
 	return tokens
 }
+
+// This returns the cyclomatic complexity for the node. This uses the method of counting one + one
+// of the following UAST Roles if present on any children:
+// If | SwitchCase |  SwitchDefault | For[Each] | [Do]While | TryCatch | Continue | OpBoolean* | Goto
+// Important: since some languages allow for code defined
+// outside function definitions, this won't check that the Node has the role FunctionDeclarationRole
+// so the user should check that if the intended use is calculating the complexity of a function/method.
+// If the children contain more than one function definitions, the value will not be averaged between
+// the total number of function declarations but given as a total.
+func CyclomaticComplexity(n *Node)  int {
+	complx := 1
+
+	iter := NewOrderPathIter(NewPath(n))
+
+	for {
+		p := iter.Next()
+		if p.IsEmpty() {
+			break
+		}
+		n := p.Node()
+		for _, r := range n.Roles {
+			switch(r) {
+			case If:
+				fallthrough
+			case SwitchCase:
+				fallthrough
+			case SwitchDefault:
+				fallthrough
+			case For:
+				fallthrough
+			case ForEach:
+				fallthrough
+			case While:
+				fallthrough
+			case DoWhile:
+				fallthrough
+			case TryCatch:
+				fallthrough
+			case Continue:
+				fallthrough
+			// FIXME: uncomment when Goto is added to the UAST
+			//case Goto:
+				//fallthrough
+			case OpBooleanAnd:
+				fallthrough
+			case OpBooleanOr:
+				fallthrough
+			case OpBooleanXor:
```

Comment:
I like your switch more. The if would be a little verbose maybe. I will modify the code and update the PR.